### PR TITLE
Awesome IPLD endeavour

### PR DESCRIPTION
**Pull request by diasdavid**

### CODE REVIEW

File: src/resolver.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  const node = util.deserialize(block.data)
+  const flatObj = flattenObject(node)
+  const paths = Object.keys(flatObj)
+                     .map((key) => {
+                       return {
+                         path: key,
+                         value: flatObj[key]
+                       }
+                     })
+  return paths
```
Comment (dignifiedquire) [inline src/resolver.js]: you can just immediately return the result, instead of creating a variable

File: src/resolver.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    if ((typeof obj[i]) === 'object') {
+      flatObject = flattenObject(obj[i])
+      for (let x in flatObject) {
+        if (!flatObject.hasOwnProperty(x)) {
+          continue
+        }
+        // Exclude arrays from the final result
+        if (flatObject[x] && Array === flatObject.constructor) {
```
Comment (dignifiedquire) [inline src/resolver.js]: you can use `Array.isArray`

File: package.json [hunk 3]
```diff
@@ -1,8 +1,8 @@
 {
-  "name": "ipld",
+  "name": "ipld-dag-cbor",
   "version": "0.6.0",
   "description": "JavaScript implementation of the IPLD (InterpPlanetary Linked Data)",
-  "main": "lib/index.js",
+  "main": "src/index.js",
```
Comment (dignifiedquire) [inline package.json]: TODO: remove before merge

File: src/resolver.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
+'use strict'
+
+const util = require('./util')
+const _times = require('lodash.times')
```
Comment (dignifiedquire) [inline src/resolver.js]: why `_times`?

File: src/resolver.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+  })
+
+  if (result) {
+    return result
+  }
+
+  // out of scope
+
+  // TODO this was my first try at writting this out of scope traversal code,
+  // it REALLY needs way more testing.
```
Comment (dignifiedquire) [inline src/resolver.js]: Is this still valid?

File: src/resolver.js [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+exports.tree = (block, options) => {
+  if (!options) {
+    options = {}
+  }
+
+  const node = util.deserialize(block.data)
+  const flatObj = flattenObject(node)
+  const paths = Object.keys(flatObj)
+                     .map((key) => {
+                       return {
```
Comment (dignifiedquire) [inline src/resolver.js]: this is some odd indenation

File: test/resolver.spec.js [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+      const paths = resolver.tree(emptyNodeBlock)
+      expect(paths).to.eql([])
+    })
+
+    it.skip('resolver.patch', (done) => {})
+  })
+
+  describe('node', () => {
+    describe.skip('resolver.resolve', () => {
+      it('path', () => {
```
Comment (dignifiedquire) [inline test/resolver.spec.js]: what about tests for these things?

File: test/resolver.spec.js [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+        value: 'baz'
+      }
+      // TODO fix array in .tree
+      /*, {
+        path: 'array/0/a',
+        value: 'b'
+      }, {
+        path: 'array/1',
+        value: '2'
+      } */])
```
Comment (dignifiedquire) [inline test/resolver.spec.js]: should be probably fixed before merge?

File: src/resolver.js (hunk 5 above)
Comment (diasdavid) [inline src/resolver.js]: It is :(

**Pull request merged**